### DISCUSSION TITLE
{"error":"Undefined array key 0"}

### DISCUSSION THREAD

**Discussion by hantsy** [symfony/symfony#51555] [Q&A]
I tried to update my example project to the latest APIs, such as used new `ValueResolverInterface` since 6.2, 

https://github.com/hantsy/symfony-rest-sample/blob/master/src/ArgumentResolver/BodyValueResolver.php

and updated custom annotation to return an array type.

```php
#[Attribute]
class Get extends Route
{
    public function getMethods(): array // add return type here.
    {
        return [HttpMethod::GET->name];
    }

}
```

And the existing tests failed.

```php

    public function testWhenGettingNonExistingPost_thenReturn404Status(): void
    {
        $client = static::createClient();
        $id = Uuid::v4();
        $crawler = $client->jsonRequest('GET', '/posts/' . $id);

        //
        $response = $client->getResponse();
        $this->assertResponseStatusCodeSame(404);
        $data = $response->getContent();
        $this->assertStringContainsString("Post #" . $id . " was not found", $data);
    }

```

Check the Github actions exception stack:

https://github.com/hantsy/symfony-rest-sample/actions/runs/6073599329/job/16475818374

```bash 
1) App\Tests\Controller\PostControllerTest::testWhenUpdatingNonExistingPost_thenReturn404Status
Failed asserting that the Response status code is 404.
HTTP/1.1 500 Internal Server Error
Cache-Control: no-cache, private
Content-Type:  application/json
Date:          Mon, 04 Sep 2023 12:46:29 GMT
X-Robots-Tag:  noindex

{"error":"Undefined array key 0"}

/home/runner/work/symfony-rest-sample/symfony-rest-sample/vendor/symfony/framework-bundle/Test/BrowserKitAssertionsTrait.php:142
/home/runner/work/symfony-rest-sample/symfony-rest-sample/vendor/symfony/framework-bundle/Test/BrowserKitAssertionsTrait.php:38
/home/runner/work/symfony-rest-sample/symfony-rest-sample/tests/Controller/PostControllerTest.php:58
```

**Comment by stof** [comment]
Well, looks like something inside the request handling triggers such an error. But as your json response does not include any information about the location of the error , there is nothing we can do to help you.

**Reply by hantsy** [reply]
You can review this change here, https://github.com/hantsy/symfony-rest-sample/commit/e3ed08e058fdece81e31986325d95709449420ce

The previous version reverted to PHPUnit 9.x and got the tests passed.
